src/__tests__/components/elements/wrapper/conditional-renderer.spec.tsx

```diff
@@ -186,6 +195,47 @@ describe("conditional-renderer", () => {
 		expect(getFieldTwo()).toBeInTheDocument();
 	});
 
+	describe.each`
+		condition        | config                                                         | invalid                 | valid
+		${"within-days"} | ${{ withinDays: { numberOfDays: 7 } }}                         | ${["09", "01", "2022"]} | ${["02", "01", "2022"]}
+		${"within-days"} | ${{ withinDays: { numberOfDays: -7 } }}                        | ${["02", "01", "2022"]} | ${["31", "12", "2021"]}
+		${"within-days"} | ${{ withinDays: { numberOfDays: 5, fromDate: "2022-01-05" } }} | ${["01", "01", "2022"]} | ${["06", "01", "2022"]}
+	`("$condition validation", ({ condition, config, invalid, valid }) => {
+		beforeEach(() => {
+			jest.spyOn(LocalDate, "now").mockReturnValue(LocalDate.parse("2022-01-01"));
+		});
+
+		afterEach(() => {
+			jest.restoreAllMocks();
+		});
+
+		it(`should support ${condition} condition for date field conditional rendering`, async () => {
+			const uiTypeField1 = "date-field";
+			const uiTypeField2 = "text-field";
+			const fields: Record<string, TFrontendEngineFieldSchema> = {
+				[FIELD_ONE_ID]: {
+					label: FIELD_ONE_LABEL,
+					uiType: uiTypeField1,
+					validation: [config],
+				},
+				[FIELD_TWO_ID]: {
+					label: FIELD_TWO_LABEL,
+					uiType: uiTypeField2,
+					showIf: [{ [FIELD_ONE_ID]: [{ filled: true }] }],
+				},
+			};
+			renderComponent(fields);
+
+			await changeDate(invalid[0], invalid[1], invalid[2]);
+			fireEvent.click(getField("button", "Done"));
+			expect(getFieldTwo(true)).not.toBeInTheDocument();
+
+			await changeDate(valid[0], valid[1], valid[2]);
+			fireEvent.click(getField("button", "Done"));
+			expect(getFieldTwo()).toBeInTheDocument();
+		});
+	});
+
```

Comment:
- wrap these tests in a new `within-days condition` test suites after `shown condition` test suite
- change existing `describe.each` to `it.each`
- update test label
- add test for date format

so it should look something like this

```
describe("shown condition", () => {
	// ...
});

// do this
describe("within-days condition", () => {
  	describe.each`
		condition                             | config                                                         | invalid                 | valid
		${"within-days (future)"}             | ${{ withinDays: { numberOfDays: 7 } }}                         | ${["09", "01", "2022"]} | ${["02", "01", "2022"]}
		${"within-days (past)"}               | ${{ withinDays: { numberOfDays: -7 } }}                        | ${["02", "01", "2022"]} | ${["31", "12", "2021"]}
		${"within-days (from specific date)"} | ${{ withinDays: { numberOfDays: 5, fromDate: "2022-01-05" } }} | ${["01", "01", "2022"]} | ${["06", "01", "2022"]}
		// add test for date format
	`("$condition validation", ({ condition, config, invalid, valid }) => {
		// ...
	});
});
```